### ISSUE TITLE
No forecast data available

### ISSUE THREAD

**Issue by mawinkler** [mawinkler/astroweather#11]
There seems to be a problem on the 7timer side accessing the API endpoints of NOAA. It is currently under investigation.
I do not have any information on when this will be fixed, but hopefully soon.

**Comment by tomlut** [comment]
Hi Markus,

The update you pushed to better handle the missing forecast seems to have had an unintended effect of generating way more errors (999 occurrences):
```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/astroweather/entity.py:33
Integration: AstroWeather (documentation, issues)
First occurred: 00:33:00 (999 occurrences)
Last logged: 08:52:04

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 735, in _update_entity_states
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 540, in async_update_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in _async_write_ha_state
    state = self._stringify_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 559, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/weather/__init__.py", line 306, in state
    return self.condition
  File "/config/custom_components/astroweather/weather.py", line 158, in condition
    return self._current.condition_plain
  File "/config/custom_components/astroweather/entity.py", line 33, in _current
    return self.coordinator.data[0]
IndexError: list index out of range
```

**Comment by mawinkler** [comment]
Hi Tom,
this bug was revealed when there was no data available at all. While fixing it the backend got alive again as well about 30mins ago. [Issue 7timer](https://github.com/Yeqzids/7timer-issues/issues/17#issuecomment-1068733317)

For now, it should be sufficient to just reload the astroweather integration and all should turn colorful on your side again. I will push the change in my entity.py with the next release to prevent the error you noted to occur again (if 7timer goes down again).

**Comment by LuckyTriple7** [comment]
Switch back to 0.20.7fixed that Problem for me.

**Comment by mawinkler** [comment]
Backend problems were resolved and seem to be stable again.

**Comment by hpservertech** [comment]
Is it not working again?  Been a few days and still showing everything is unavailable.  tried reloading the integration as well as rebooting the server and it didn't help like it did a couple weeks ago.

**Comment by mawinkler** [comment]
Hm, I'm not seeing any errors and it works flawlessly on my side. Do you have any logs to provide?

**Comment by hpservertech** [comment]
Where can I get the logs?

**Comment by mawinkler** [comment]
in home-assistant.log or via the frontend Configuration --> Server Controls --> Logs

**Comment by mawinkler** [comment]
Hi, can it be, that at the time the error occurred you had very bad conditions? If so, I just released a new card which fixes a bug in the detection of the weather entity. Give it a try.

**Comment by hpservertech** [comment]
Actually everything went unavailable and it added new devices but it didn't add all of them.  I look at the forcast today & tomorrow the most since it is a simple gauge but noticed on the card today that it's completely wrong.  It's raining all day and it shows 0% clouds.  Not sure how to add a screenshot but here is a link to one I just took.
https://photos.app.goo.gl/YqydWQ4Hbc1EUAJN7

I downloaded & added the card yesterday afternoon (cst), is a newer one available?

**Comment by mawinkler** [comment]
It shows 0% cloudless, so fully clouded. There should be a new version of the card available after your next update run of hacs

**Comment by hpservertech** [comment]
ok, so then the chart in the middle is actually CloudLESS not Clouds as it says..correct?

Any way to get the forcast today & tomorrow working again?

**Comment by mawinkler** [comment]
As I can see the forecast today and tomorrow seem to work on your screenshot?? It looks pretty the same as on my screenshot here [card](https://github.com/mawinkler/astroweather-card). What are you missing?

**Comment by mawinkler** [comment]
Question: How did you build your allsky camera? Is it self-made or a commercial one?

**Comment by hpservertech** [comment]
Self made.  Running on a Pi using an ASI120mc inside an old CCTV dome camera housing.  Attaches to a pole that goes to a box where the Pi, & relays are located.  Has 2 5w ceramic resisters mounted in the dome for dew/heat.  Also have DHT22 sensors in the dome that will trigger the relays to turn on/off.  I need to do some changes inside the dome to get rid of the red paint reflection from the camera.  you should be able to see it at http://allsky.duckdns.org

I manually set the status but after reboot those 2 went back to unavailable.

**Comment by hpservertech** [comment]
Guess I will end up using the deepsky forcast.  The higher the percentage the better the viewing correct?

The cloudless is confusing.  Shows 12% but then Total Cloud Cover less then 20% doesn't make sense.  I read cloudless 12% meaning there is 82% cloud cover.  Getting hung up on the words I think more then anything and not understanding them.

![Screenshot 2022-04-05 154708](https://user-images.githubusercontent.com/85696789/161846284-8f637bb9-89b0-4718-aee4-87e2601589a6.png)

**Comment by hpservertech** [comment]
Think there is a TOD issue.  You can see from my last screenshot that tomorrow looked better then today, but tonight at 10pm CST it says today is at 76% and tomorrow is bad.  Happened to notice this when going out to cover the scope up for the day.

![Screenshot 2022-04-05 220326](https://user-images.githubusercontent.com/85696789/161887721-d678ae0a-74a6-4b12-8a71-f58da5991a71.png)

**Comment by mawinkler** [comment]
So, the logic is as follows: I'm calculating the condition based on transparency, seeing, and cloud cover. When you think about the influence of these measurements it obviously means the lower the measurement is, the better for viewing conditions. This also means if I want to calculate an easy-to-read overall condition value, which at its max has the highest value, I need to invert the measurements.
In the top right of your screenshot you see the values in bolt Condition, **Cloudless**, Seeing, and Transparency. Condition is calculated based on the other three.
In the table below, you have actually the same values, the current and the forecast.

**Comment by mawinkler** [comment]
Flip

> Guess I will end up using the deepsky forcast. The higher the percentage the better the viewing correct?
> 
> The cloudless is confusing. Shows 12% but then Total Cloud Cover less then 20% doesn't make sense. I read cloudless 12% meaning there is 82% cloud cover. Getting hung up on the words I think more then anything and not understanding them.
> 
> ![Screenshot 2022-04-05 154708](https://user-images.githubusercontent.com/85696789/161846284-8f637bb9-89b0-4718-aee4-87e2601589a6.png)

Oh, the top part of the card displays the current situation.
The second shows the forecast for this evening and tomorrow evening in plain text.
The lower part is the forecast in a three hour interval.

The evening forecasts for Today and Tomorrow are calculated on the forecast based using your local time and a timeframe of 9hs.
From a comment In my code:
```
# between 9pm to 3am.
# Possible timestamps within the data:
# 15 18 (21 00 03) 06 09 12
# 16 (19 22 01) 04 07 10 13
# 17 (20 23 02) 05 08 11 14
# Relevant ones in brackets
```
Since I'm only having data based on three hours intervals the time interval for the calculation might start at 7pm, 8pm, or 9pm depending on your timezone. If it's now for example 10pm your local time, I miss the start for today and flip over to tomorrow and the day after. I know that this is not perfect but for the current night you don't need a forecast anymore. I think the best would be to attach a date instead `today` or `tomorrow`.

**Comment by mawinkler** [comment]
What do you think about this?
<img width="1016" alt="aw-prop-1" src="https://user-images.githubusercontent.com/16704258/161924136-4f36038c-7115-48e6-93e8-b61a25e4d2b3.png">

**Comment by hpservertech** [comment]
Think it looks good.  Some feedback
Cloudless - Still doesn't make sense.  In the US we just dont refer to things in this way.    We look at things as how much cloud cover there is, not now much there isn't
Chart - Without hovering over the points there is no way to know what each of those lines are.  Think the colors of the lines should match the color of text above it or have labels on the chart.

Would still like a simple gauge like you had before as an entity for the forecast.  The deep sky forecast will work but for simple quick look the gauge is what I looked at 99% of the time.

Here is another project I did.  Something cheaper then Bluetti or Jackery.  I have the LCD also sending data out via MQTT as well now and eventually looking at building a plugin for Nina to show battery status and input into the advance sequence.  https://diysolarforum.com/threads/astrophography-build.36487/

**Comment by mawinkler** [comment]
I totally agree, cloud coverage is much more common, even here in Europe. Using cloud coverage would then mean, that the line in the chart would be in the top area if there are many clouds, but this would be bad for the viewing conditions of course. I was thinking more of a way of creating a graphical display that directly shows the viewing conditions' quality. Same for the distinct values. If cloudless is 100%, this means from the perspective of the clouds my condition will be good. Compared to that, if I'd use cloud cover, it would mean I'm going to stay inside today. If that makes kind of sense.

Line colors agreed. I had that in the beginning, but ppl are using different skins and some combinations look very weird. So I just stuck to the default and highlight color for now.

For the gauge, you can still do that. Just use the sensors:

- sensor.astroweather_deepsky_forecast_today and
- sensor.astroweather_deepsky_forecast_tomorrow

<img width="546" alt="bars" src="https://user-images.githubusercontent.com/16704258/162022340-a5f3228c-34b9-4fa2-ab99-49c5aa8cf997.png">

with this yaml code:

```yaml
  - type: entities
    entities:
      - positions:
          minmax: inside
        entity_row: true
        min: 0
        max: 100
        target: 80
        name: Today
        type: custom:bar-card
        entities:
          - entity: sensor.astroweather_deepsky_forecast_today
      - positions:
          minmax: inside
        entity_row: true
        min: 0
        max: 100
        target: 80
        name: Tomorrow
        type: custom:bar-card
        entities:
          - entity: sensor.astroweather_deepsky_forecast_tomorrow
```

**Comment by mawinkler** [comment]
All good :-) I really like any kind of constructive feedback, no doubt.
I just pushed some changes to the repos. I didn't do new releases until now, but if you redownload with hacs from the main branch you'll have the latest version of the card and the integration itself. So you need to reload both from main. It's still using cloudless, but your other proposals are in.

Your level of AP sounds impressive! I just started with that hobby about 15 months ago but I'm now totally addicted to it. I wish we had more clear skies here in Bavaria, though. Are you sharing your photographs (e.g. on Astrobin) or having a web page?

**Comment by hpservertech** [comment]
I only started last summer myself.  Right now just have some stuff in my
google photos.  Here is a link to some.  That red thing is my arduino based
led light panel, 3d printed.  The Bodes Galaxy is one of the best images I
have done so far with my QHY camera.  The first few images were taken back
when I was just using my Nikon cameras & lenses.  Moved up to an 80mm
tripplet now and have an 8" sct for planetary (for the moment)

https://photos.app.goo.gl/P7eHuZckqBTNTenx8
